### PR TITLE
Add a PhoneNumber class to handle formatting and actions on numbers

### PR DESCRIPTION
**Pull request by caitlinrhd**
We value collaboration at Invoca, and focus on peer mentorship as a way for the whole team to learn and thrive. As part of this process, we conduct code reviews during our development cycle. In this phase of the interview, we would like you to review and comment on this Pull Request. 

You can assume that this Pull Request was written by an intern, who is new to professional software development.  Please take about 20-30 minutes to review this Pull Request and comment as if you were reviewing an intern's code.

### CODE REVIEW

File: phone_number.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    open_paren = @number.index('(')
+    close_paren = @number.index(')')
+    if open_paren
+      if close_paren
+        @number[(open_paren + 1)..(close_paren - 1)]
+      end
+    else
+      @number.gsub!(/[^0-9]/, "")
+      @number[0..2]
+    end
```
Comment (sjhewitt) [inline phone_number.rb:50]: there's a subtle bug here if the input string contains `(` but not `)` e.g. `(100-100-1000`, then the method would not return the expected area code

File: phone_number_spec.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
+require_relative "./phone_number"
+
+RSpec.describe PhoneNumber do
+  let(:phone_number) { PhoneNumber.new("5558675309") }
```
Comment (sjhewitt) [inline phone_number_spec.rb:4]: Can we add some tests to check the edge cases of phone number initialization:
e.g.
```
PhoneNumber.new("55-58-67-53-09")
PhoneNumber.new("1")
PhoneNumber.new("132165876321632131926123672317861236783126713261326798312")
PhoneNumber.new("thisisnotanumber")
```

**Review by sjhewitt — CHANGES_REQUESTED** [review]
Looks like a good start! I spotted a small bug which needs addressing, and I think the test suite should be expanded to test more inputs for the PhoneNumber constructor
